### PR TITLE
feat: add build to generate web index of official notebooks

### PR DESCRIPTION
**Pull request by andrewferlitsch**
**REQUIRED:** Add a summary of your PR here, typically including why the change is needed and what was changed. Include any design alternatives for discussion purposes.

<br>
Add cloudbuild script to generate searchable index of official notebooks for the webdocs
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [ ] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [ ] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [ ] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [ ] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: .cloud-build/notebook-index-cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  # Install Python dependencies and run index generation script
+  - name: ${_PYTHON_IMAGE}
+    entrypoint: /bin/sh
+    args:
+    - -c
+    - |
+      . workspace/env/bin/activate &&
+      python3 notebooks/notebook_template_review.py --web --title -steps --desc --linkback --notebook-dir=notebooks/official
+    env:
+    - 'IS_TESTING=1'
```
Comment (ivanmkc) [inline .cloud-build/notebook-index-cloudbuild.yaml:32]: It this used somewhere?

File: .cloud-build/notebook-index-cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+steps:
+  # Show the gcloud info and check if gcloud exists
+  - name: ${_PYTHON_IMAGE}
+    entrypoint: /bin/sh
+    args:
+    - -c
+    - gcloud config list --quiet
```
Comment (ivanmkc) [inline .cloud-build/notebook-index-cloudbuild.yaml:7]: This is just for debugging so you can remove it.

File: .cloud-build/notebook-index-cloudbuild.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  # Check the Python version
+  - name: ${_PYTHON_IMAGE}
```
Comment (ivanmkc) [inline .cloud-build/notebook-index-cloudbuild.yaml:9]: Ditto.

File: .cloud-build/notebook-index-cloudbuild.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    - -c
+    - python3 -m venv workspace/env
+  # Install Python dependencies and run index generation script
+  - name: ${_PYTHON_IMAGE}
+    entrypoint: /bin/sh
+    args:
+    - -c
+    - |
+      . workspace/env/bin/activate &&
+      python3 notebooks/notebook_template_review.py --web --title -steps --desc --linkback --notebook-dir=notebooks/official
```
Comment (ivanmkc) [inline .cloud-build/notebook-index-cloudbuild.yaml:30]: This just writes in the filesystem of the test runner container so you won't be able to access it. You have to store somewhere in the cloud to access it.

Consider: https://cloud.google.com/build/docs/building/store-artifacts-in-cloud-storage

**Review by ivanmkc — APPROVED** [review]

File: .cloud-build/notebook-index-cloudbuild.yaml (hunk 2 above)
Comment (andrewferlitsch) [reply .cloud-build/notebook-index-cloudbuild.yaml:7]: removed

File: .cloud-build/notebook-index-cloudbuild.yaml (hunk 1 above)
Comment (andrewferlitsch) [reply .cloud-build/notebook-index-cloudbuild.yaml:32]: removed

File: .cloud-build/notebook-index-cloudbuild.yaml (hunk 3 above)
Comment (andrewferlitsch) [reply .cloud-build/notebook-index-cloudbuild.yaml:9]: removed

**Pull request merged**